### DISCUSSION TITLE
Receive MQTT messages in a reactive notebook with Mosquitto.jl

### DISCUSSION THREAD

**Discussion by s-celles** [fonsp/Pluto.jl#2834] [Questions about using Pluto]
Hello,

following https://github.com/denglerchr/Mosquitto.jl/issues/15#issuecomment-1973324835 I'm trying to turn this [Mosquitto.jl example](https://github.com/denglerchr/Mosquitto.jl/blob/main/examples/03_subscribe_onconnect.jl) into a reactive notebook.

What this example does?
It subscribes to every `test/...` topics at test.mosquitto.org and displays messages on console. (limited to 20 incoming messages)

What I want to achieve?
I want to make a reactive notebook (using Pluto.jl) which also subscribe to every `test/...` topics at test.mosquitto.org and store messages and their topic into a DataStructures CircularBuffer. This circular buffer is displayed in the notebook. The reception should be always running.

What I did?
I created a Ratio (PlutoUI) which can change running_state from `"stopped"` to `"running"`

What is wrong?
The cell containing `loop_forever2` is always running when running_state is running but other cell are not being updated, especially one which should display circular buffer content.

I have no idea what is wrong with my notebook.

Some help will be very nice.

Kind regards

**Comment by Pangoraw** [comment]
For this use case of asynchronous polling, I would recommend using the [PlutoHooks.jl](https://github.com/JuliaPluto/PlutoHooks.jl) package. here is an example of triggering reactive runs from ZMQ messages which should not be too different with Mosquitto: https://pangoraw.github.io/Notebooks/zmq_socket
